lib/directives.js

```diff
@@ -8,4 +8,20 @@ const getImportedDirectives = function (component) {
   return Object.assign({}, component._directives, ...component._importedDirectives);
 };
 
-module.exports = { mergeDirectives, getImportedDirectives };
+const namespaceDirectivesInTypeDefs = (types = [], id) => {
+  const regex = /(@[a-zA-Z0-9]*)\b/g;
+  return types.map(t => t.replace(regex, `$1_${id}`));
```

Comment:
Types may not be a string - it may be a document. We really can only perform this renaming at the document level — probably just prior to schema creation.